### ISSUE TITLE
[PERF] Issue Title

### ISSUE THREAD

**Issue by NewtSoup** [ValveSoftware/SteamVR-for-Linux#613]
**Describe the bug**
Since the introduction of  Steam VR 1.27 Major Update  I have been experiencing massive latency and reprojection in the headseat to the point where most VR games are now unplayable.   The Steam VR performance graph reads 28.9 of 11.2 ( I am running my headset at 90hz ) 

**To Reproduce**
1. Go to the steam icon in the taskbar
2. Click on the icon and select steam vr
3. Start any VR game
4. See image ghosting in game and the feeling of it runnint at very low fps when it's not and experience lagging view update when you turn your head.  Moving hands or visible weapons is smooth.   I have tested in both Fedora and Ubuntu and experience the same issue.

**Expected behavior**
Was hoping the Major Update would improve issues ( though one issue HAS been cured - I no longer get a system lock up in Subnautica at the North Island.

**System Information (please complete the following information):**
- Fedora/Nobara and Ubuntu
- SteamVR version: 1.27
- Steam client version: 1694466342
- Opted into Steam client beta?: No  [ though  currently beta seems to be the same as release on trying  ]
- Graphics driver version:   Mesa 23.3.0-devel (LLVM 16.0.6)
- Gist for SteamVR System Information: https://gist.github.com/NewtSoup/efb00bbf3db9f0b7f211dc42803f1d15.js

**Performance Data (optional, but very helpful):**
- CPU: i7-7700
- GPU: RX6700XT Navi22
- Gist for perf top: [See [instructions](https://github.com/ValveSoftware/SteamVR-for-Linux/blob/master/CapturingPerfData.md)]
- Link to gpuvis trace: [See [instructions](https://github.com/ValveSoftware/SteamVR-for-Linux/blob/master/CapturingPerfData.md)]

**Screenshots**
Screenshot of VR Performance graph showing 29 of 11.1 ms for Fallout 4.  The effect cannot be captured on video because while the headset display has poor performance the on screen display is smooth.
https://i.imgur.com/qyGVdGQ.png

**Additional context**
I was getting very acceptable performance in 1.26 apart from random lockups that would require a system reset to recover from.   The system lock up was reliably reproduced by visiting the north island in subnautica while playing in VR.  The latest 1.27 version of Steam VR seems to have fixed the issue as I can visit the north island now but the game is not playable for more than 5 minutes due to the performance issues introduced by 1.27.   I have tested the performance in both  Fedora ( Nobara ) and Ubuntu and the issues are present in both OS.

As an interim measure for those users experiencing these problems ( the Steam Discussion is full of complaints ) could we perhaps have the option to roll back to 1.26 under the "Beta's" tab for Steam VR?  Indeed could the option to roll back a version or two be "a thing" moving forward?

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" and "Perf Data" sections in their replies.

**Comment by NewtSoup** [comment]
Poor performance is also present in  Steam VR  Beta 2.0

**Comment by VRcontrib** [comment]
I'm having the same issue.

- Arch Linux
- SteamVR version: 1.27 or 2.0
- Opted into Steam client beta?: Yes
- Ryzen 7 3700X
- NVIDIA GeForce 2060

**Comment by NewtSoup** [comment]
The following command has improved things a little for me:

echo "high" | sudo tee  /sys/class/drm/card1/device/power_dpm_force_performance_level

Depending on your system card1 may be card0 or perhaps even card2.  For me it's card1 because I have an Intel system and card0 is the onboard GPU ( I think that's why ).

This forces the GPU into high power mode and makes UltraWings 2 smooth again 8.9/11ms

It brings FO4VR down to 14.9/11ms which is smoother but still has a  lot of reprojection and is not really playable.

This command needs issuing after every boot unless you use a udev rule to make it permanent.

**Comment by VRcontrib** [comment]
Thank you for your response!

I'm getting Permission Denied even when using sudo with the command, so I assume my card doesn't support that.

**Comment by VRcontrib** [comment]
Thank you for your response!

I'm getting Permission Denied even when using sudo with the command, so I assume my card doesn't support that, unfortunately. Do you have any other ideas on how that would be done with other cards? How did you figure out that you should do that in the first place?

**Comment by NewtSoup** [comment]
> Thank you for your response!
> 
> I'm getting Permission Denied even when using sudo with the command, so I assume my card doesn't support that, unfortunately. Do you have any other ideas on how that would be done with other cards? How did you figure out that you should do that in the first place?

try doing sudo -i on it's own first  then run the command.  It's likely a permission issue with the shell.

**Comment by Taiko2k** [comment]
Also having the issue. Since SteamVR 1.27 there's a double vision effect any time re-projection kicks in.

**Comment by NewtSoup** [comment]
Having switched to the steam client beta and steam vr beta I can confirm that poor performance is still present in Beta 2.02.

**Comment by razgrim** [comment]
Only just got my headset so can't confirm this is NEW issue, but
It's playable, but it isn't enjoyable. Default, headset powered on, dark mountain range no problems. Soon as any VR app is loaded (the ones that do), Turning head becomes jumpy slideshow like the image is shimmying or something. Very rough start with a big purchase. 

Re-installed everything to clear up possible issues. 
Manjaro latest fresh install, kernel 6.5.5-1
Everything updated to latest
AMD Radeon RX 6700 XT
AMD Ryzen 7 2700X
32GB RAM

It ain't hardware brudda

**Comment by NewtSoup** [comment]
> Only just got my headset so can't confirm this is NEW issue, but It's playable, but it isn't enjoyable. Default, headset powered on, dark mountain range no problems. Soon as any VR app is loaded (the ones that do), Turning head becomes jumpy slideshow like the image is shimmying or something. Very rough start with a big purchase.
> 
> Re-installed everything fresh OS up, to clear up possible issues. Valve Index, updated Manjaro latest fresh install, kernel 6.5.5-1 Everything updated to latest AMD Radeon RX 6700 XT AMD Ryzen 7 2700X 32GB RAM
> 
> It ain't hardware brudda
> 
> Tried doing a manual downgrade to SteamVR 1.26.7 but doesn't appear to be trivial, missed something because that vers wasn't recognizing headset. Would love a proper "beta" menu option for 1.26.

No, it's not hardware.  Mine behaves fine in Windows.  It's the janky linux drivers that are being released untested.
It would be great if Steam VR for Linux were made open source so that the community could contribute.  Or at least the Valve Index telemetry data so that an open source Open XR driver for the headset. could be implemented.

**Comment by NewtSoup** [comment]
Update: 2.0.10 

With this update I am seeing some improvement!  It's still not back to the performance it was  in 1.26 but I now see the new interface and the frame timings are mostly within the 11.1ms @ 90hz.

- Beat Saber - As expected.  Good Frame Timing,  few to no magenta bars
- VR Regatta - As Expected  Good Frame Timing , few to no magenta bars
- V-Racer Hoverbike - As expected,  Good Frame Timing, few to no magenta bars
- Skyrim VR - Playable - Some minor performance issues with magenta bars, frame timing is good.
- Ultrawings 2 - Playable, some minor performance issues with magenta bars, frame timing is good.
- Half Life VR Mod - Unplayable - good frame timing but a LOT of magenta bars causing double vision.
- Fallout 4 VR - Unplayable - Poor Frame Timing 19ms @90hz, lots of magenta bars.
- Subnautica - Unplayable - Terrible frame timings  25ms @ 90hz and a WALL of magenta appears when on land but the game no longer crashes at the north island.  https://i.imgur.com/Gt9Z5bx.png

- Half Life Alyx - Not tested
- Elite Dangerous  - not tested
- No Man's Sky - not tested

All of the above games worked "As Expected" with only Fallout 4 VR in the "Playable" category at 1.26 - But even that looked nice, sometimes I would get double vision but then restarting Steam VR would usually fix it.

Can anyone explain what the magenta bars mean?   Someone told me they are CPU late starts but I still don't know if that's correct or what it actually means in terms of performance.

**Comment by dagelf** [comment]
I changed my window manager to `icewm` and all the dropping went away. Went from unusable to perfect.
